services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java

```diff
@@ -601,6 +600,10 @@ public void start() throws VeniceException {
       service.start();
     }
 
+    for (AbstractStorageEngine storageEngine: storageService.getStorageEngineRepository().getAllLocalStorageEngines()) {
```

Comment:
I have moved the function here which is after services start, so that `HelixParticipationService` is started/initialized for referencing the ideal state in function.
I have reorganized the code in `StorageService` to its initial state.

Reply:
It looks like `HelixParticipationService` is started asynchronously, and it is causing some tests to fail.

We can try using [managerFuture](https://github.com/linkedin/venice/blob/df1c0b9f3b19a7dc4fbd3113512e3a21abbc278f/services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java#L356C3-L356C83). This will runs things after `HelixParticipationService` has started.

Ex:
```
managerFuture.thenApply(manager -> {
      // Call checkWhetherStoragePartitionsShouldBeKeptOrNot here
});
```